images/inference-services/kfserving/run.sh

```diff
@@ -65,9 +106,9 @@ printf "${prediction_url}" > /tekton/results/${TASK_RESULT}
 
 inside_metadata=""
 first_resource=1
-envsubst < /root/template.sh | while read line
+envsubst < /root/template.sh | while read -r line
 do
-  if echo $line | grep -q '^[ ]*kind'; then
+  if echo "$line" | grep -q '^[ ]*kind'; then
```

Comment:
could you please move this whole logic out of `run.sh` into a common library, so we can reuse it in other predictors ? it's a great way to convert a list of k8s manifests and convert them into a list of name/kind resources. I, for one, could definitely use it in my OVMS image :)

Reply:
please disregard this comment. I'd like to do this myself, given that I'm already putting together a base image based on your recommendation :)

Reply:
perfect 👍